### DISCUSSION TITLE
Disable line wrapping or only show the first x characters of a long text

### DISCUSSION THREAD

**Discussion by bookzhan** [josdejong/svelte-jsoneditor#363] [Ideas]
If line breaks are used to display the content, it would be inconvenient to view the structure of the JSON.
<img width="818" alt="image" src="https://github.com/josdejong/svelte-jsoneditor/assets/11169519/b5dda7b2-d066-437c-a1ab-936f0c0d3217">

**Comment by josdejong** [comment]
Thanks for your suggestion. I can think of two solutions:

1. Introduce an option "lineWrap" that you can turn on/off yourself.
2. Only show the first say 100 characters, followed by a button `[...]` which, when clicked, will show the remaining characters. This is how browser consoles do it, and I think that is a much nicer solution since you don't have to toggle options upfront. We could apply this to the `tree` and `table` mode too.

Anyone interested in thinking this through and implement a solution? Help would be welcome.

**Comment by josdejong** [comment]
Moving this idea to the [Discussions > Ideas](https://github.com/josdejong/svelte-jsoneditor/discussions/categories/ideas) section since we are not actively working on it. Help implementing this would be very welcome.
